mlir/lib/Transforms/Utils/DialectConversion.cpp

```diff
@@ -761,7 +721,8 @@ class IRRewrite {
     MoveBlock,
     SplitBlock,
     BlockTypeConversion,
-    MoveOperation
+    MoveOperation,
+    ModifyOperation
```

Comment:
I know this list isn't ordered ... not sure if there is a good reason why it isn't :)

Reply:
That's so that I can implement `classof` with comparisons:
```c++
  static bool classof(const IRRewrite *rewrite) {
    return rewrite->getKind() >= Kind::MoveOperation &&
           rewrite->getKind() <= Kind::ModifyOperation;
  }
```

Block rewrites must appear before operation rewrites.

Reply:
Ah, could you document that here?